Sources/StreamVideoSwiftUI/CallView/VideoParticipantsView.swift

```diff
@@ -99,47 +119,185 @@ public struct VideoCallParticipantModifier: ViewModifier {
                                 isPinned: participant.isPinned
                             )
                             
+                            Spacer()
+
                             if showAllInfo {
-                                Spacer()
                                 ConnectionQualityIndicator(
                                     connectionQuality: participant.connectionQuality
                                 )
                             }
                         }
-                        .padding(.bottom, 2)
                     })
-                    .padding(.all, showAllInfo ? 16 : 8)
-                    
-                    if participant.isSpeaking && participantCount > 1 {
-                        Rectangle()
-                            .strokeBorder(Color.blue.opacity(0.7), lineWidth: 2)
-                    }
-                    
-                    if popoverShown {
-                        ParticipantPopoverView(
-                            participant: participant,
-                            call: call,
-                            popoverShown: $popoverShown
-                        )
-                    }
                 }
             )
-            .onTapGesture(count: 2, perform: {
-                popoverShown = true
-            })
-            .onTapGesture(count: 1) {
-                if popoverShown {
-                    popoverShown = false
-                }
-            }
+            .applyDecorationModifierIfRequired(
+                VideoCallParticipantOptionsModifier(participant: participant, call: call),
+                decoration: .options,
+                availableDecorations: decorations
+            )
+            .applyDecorationModifierIfRequired(
+                VideoCallParticipantSpeakingModifier(participant: participant, participantCount: participantCount),
+                decoration: .speaking,
+                availableDecorations: decorations
+            )
+            .clipShape(RoundedRectangle(cornerRadius: 16))
+            .clipped()
     }
-    
+
     @MainActor
     private var participantCount: Int {
         call?.state.participants.count ?? 0
     }
 }
 
+extension View {
+
+    @ViewBuilder
+    public func applyDecorationModifierIfRequired<Modifier: ViewModifier>(
```

Comment:
This helper method allows to add decorations to the participantView. Decorations can be one of the following 1. options menu, 2. speaking stroke. In this way, in our demo app we can render the participantView without the optionsMenu decoration which we later enrich with the Stats option and add it to the topLeading corner